### PR TITLE
Implement code changes to enhance functionality and improve performance

### PR DESCRIPTION
**Pull request by MOJOliciousFTW**
## Description

This pull request introduces a complete solution for training a machine learning model to predict the target variable, as well as updates the project dependencies to support the workflow. The main changes are the addition of a full pipeline in `main.py` for data loading, preprocessing, model training, validation, and submission, along with the specification of required libraries in `pyproject.toml`.

Model development and workflow implementation:

* Added a full machine learning workflow to `main.py`, including data loading, exploratory data analysis, categorical encoding, missing value handling, train/validation split, Random Forest model training, validation scoring, test prediction, and submission file creation.

Dependency management:

* Updated `pyproject.toml` to include essential libraries (`pandas`, `numpy`, `scikit-learn`, `matplotlib`) required for data processing and modeling.

## Additional information
- [ ] Issues this PR closes are linked under development --->
- This PR is related to issue: 

## Types of changes
- [ ] We use labels --->

## Checklist:
- [x] Works on my PC!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1 +1,71 @@
...
+# Features and target
+X = train_df.drop(["id", "y"], axis=1)
+y = train_df["y"]
+X_test = test_df.drop(["id"], axis=1)
+
+# Split for validation
+X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.2, random_state=42)
+
+# Model
+model = RandomForestClassifier(n_estimators=100, random_state=42)
```
[the remark this answers is not in the document]
Comment (MOJOliciousFTW) [reply main.py:58]: fast enough now.

File: main.py [hunk 2]
```diff
@@ -1 +1,71 @@
...
+    "loan",
+    "contact",
+    "month",
+    "poutcome",
+]
+le = LabelEncoder()
+for col in categorical_cols:
+    if col in train_df.columns:
+        train_df[col] = le.fit_transform(train_df[col].astype(str))
+        test_df[col] = le.transform(test_df[col].astype(str))
```
[the remark this answers is not in the document]
Comment (MOJOliciousFTW) [reply main.py:42]: there are no more training sets, just one.
